kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/json/JsonRecordWriterProvider.java

```diff
@@ -76,6 +77,18 @@ public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final Stri
         public void write(SinkRecord record) {
           log.trace("Sink record with view {}: {}", recordView, record);
           try {
+            // headers need to be enveloped for json format
+            if (recordView instanceof HeaderRecordView) {
+              byte[] rawJson = converter.fromConnectData(
+                  record.topic(),
+                  recordView.getViewSchema(record, true),
+                  recordView.getView(record, true)
+              );
+              s3outWrapper.write(rawJson);
+              s3outWrapper.write(LINE_SEPARATOR_BYTES);
+              return;
+            }
+
             Object value = recordView.getView(record, false);
```

Comment:
boolean envelope = recordView instanceof HeaderRecordView;
```suggestion
            Object value = recordView.getView(record, envelope);
```

why can't we do something like this?

Reply:
Sounds good.